### PR TITLE
[exporter/clickhouse] moving database creation prcocess to a start function

### LINKED ISSUE #17558: [exporter/clickhouse] moving database creation prcocess to a start function
### Component(s)

exporter/clickhouse

### Describe the issue you're reporting

origin :https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16477/files#r1051319797

moving all database creation(trace,log,metrics) prcocess to a start function.

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** moving database creation prcocess to a start function

resolve https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17558

### CODE REVIEW

**Comment by Frapschen** [comment]
need skipchangelog label

**Comment by dmitryax** [comment]
> need skipchangelog label

You can add `[chore]` tag to the PR title to get the same effect

**Comment by dmitryax** [comment]
@Frapschen please resolve the linter issues

File: exporter/clickhouseexporter/config.go [hunk 1]
```diff
@@ -58,13 +58,15 @@ type QueueSettings struct {
...
+
 var (
 	errConfigNoEndpoint      = errors.New("endpoint must be specified")
 	errConfigInvalidEndpoint = errors.New("endpoint must be url format")
 )
 
-// Validate validates the clickhouse server configuration.
-func (cfg *Config) Validate() (err error) {
+// validate the clickhouse server configuration.
+func (cfg *Config) validate() (err error) {
```
Comment (dmitryax) [inline exporter/clickhouseexporter/config.go:69]: If you keep it exported, you don't need to call it explicitly.

Comment (Frapschen) [reply exporter/clickhouseexporter/config.go]: you are right, I recover it.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Please fix the linter issues

**Comment by Frapschen** [comment]
> Please fix the linter issues

the lint  rule https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4102199457/jobs/7074888900#step:9:97 a little bother me, I find it's sad that I need  call Validate() explicitly to avoid this lint failure.

**Comment by Frapschen** [comment]
not relate ut fail
```
FAIL
make[2]: *** [../../Makefile.Common:98: test] Error 1
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver'
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make[1]: *** [Makefile:167: receiver/sshcheckreceiver] Error 2
make: *** [Makefile:97: gotest] Error 2
```

**Comment by Frapschen** [comment]
@dmitryax  ready to merge :)

**Review by dmitryax — APPROVED** [review]

**Pull request merged**